crates/shared/src/order_validation.rs

```diff
@@ -157,6 +159,12 @@ impl From<SignatureValidationError> for ValidationError {
     }
 }
 
+#[mockall::automock]
+#[async_trait]
+pub trait LimitOrderCounting: Send + Sync {
+    async fn count(&self, owner: H160) -> Result<u64>;
```

Comment:
nit: this function name seems rather unspecific.

Reply:
I see your point with the trait name but you ended up not calling the method through the trait so you would have to use an IDE to quickly recover this context. `count_limit_orders()` would have enough information even if you don't know that the trait exists.